src/Utils/utils.ts

```diff
@@ -561,3 +564,30 @@ export function omitBy<T extends Record<string, unknown>>(
     Object.entries(obj).filter(([_, value]) => !predicate(value)),
   ) as Partial<T>;
 }
+
+export const useClipboard = () => {
+  const [copied, setCopied] = useState(false);
+
+  const copyToClipboard = async (
+    text: string,
+    successMsg: string,
+    errorMsg: string,
+    iconResetDuration = 2500,
+  ) => {
+    if (!text) {
+      Notification.Error({ msg: "Nothing To Copy" });
+      return;
+    }
+
+    try {
+      await navigator.clipboard.writeText(text);
+      setCopied(true);
+      Notification.Success({ msg: successMsg });
+      setTimeout(() => setCopied(false), iconResetDuration);
+    } catch (err) {
+      Notification.Error({ msg: errorMsg });
+    }
+  };
+
+  return { copied, copyToClipboard };
+};
```

Comment:
i don't see any reason why this needs to be a hook. Let's keep it as a simple fn. instead

Reply:
> i don't see any reason why this needs to be a hook. Let's keep it as a simple fn. instead

I figured that out, made the required changes @rithviknishad